pkg/apis/imageregistry/v1/types.go

```diff
@@ -208,21 +201,26 @@ type ImageRegistryConfigStorageSwift struct {
 	TenantID  string `json:"tenantID"`
 }
 
-type ImageRegistryConfigStorageFilesystem struct {
-	VolumeSource corev1.VolumeSource `json:"volumeSource"`
-}
-
 type ImageRegistryConfigStoragePVC struct {
 	Claim string `json:"claim"`
 }
 
+// ImageRegistryConfigStorage describes how the storage should be configured
+// for the image registry.
 type ImageRegistryConfigStorage struct {
-	Azure      *ImageRegistryConfigStorageAzure      `json:"azure,omitempty"`
-	Filesystem *ImageRegistryConfigStorageFilesystem `json:"filesystem,omitempty"`
-	GCS        *ImageRegistryConfigStorageGCS        `json:"gcs,omitempty"`
-	S3         *ImageRegistryConfigStorageS3         `json:"s3,omitempty"`
-	Swift      *ImageRegistryConfigStorageSwift      `json:"swift,omitempty"`
-	PVC        *ImageRegistryConfigStoragePVC        `json:"pvc,omitempty"`
+	// EmptyDir represents an experimental storage that uses a temporary
+	// directory that shares a pod's lifetime. This storage is not suitable
+	// for production use. It can't be used with more than 1 replica. When the
+	// pod is removed from a node for any reason, the data in the emptyDir is
+	// deleted forever.
+	EmptyDir *ImageRegistryConfigStorageEmptyDir `json:"emptyDir,omitempty"`
```

Comment:
Suggested godoc:

```suggestion
	// EmptyDir represents ephemeral storage on the pod's host node.
	// This storage cannot be used with more than 1 replica and is not suitable
	// for production use. When the pod is removed from a node for any reason, 
	// the data in the emptyDir is deleted forever.
	EmptyDir *ImageRegistryConfigStorageEmptyDir `json:"emptyDir,omitempty"`
```

Reply:
You've dropped "experimental" from the description. This storage may be removed from the next releases (or we may want to remove it right now).